### PR TITLE
Simpler tiebreaker function for resolve duplicates

### PR DESCRIPTION
**Pull request by joeranbosma**
- Minor cleanup

It's longer, but easier to follow IMO. Let me know what you think.

Note: pytest now fails, so I switched something somewhere. Will investigate if you like the structure.

### CODE REVIEW

File: src/picai_prep/dcm2mha.py [hunk 1]
```diff
@@ -277,44 +275,62 @@ def apply_mappings(self):
...
-        # value_func, largest, msg = tiebreaker
-        tiebreaker_slice_count = ('slice count', lambda a: len(a.filenames), True)
-        tiebreaker_image_resolution = ('image resolution', lambda a: a.resolution, False)
-
-        # create dict collecting all items for each mapping
-        for i, serie in enumerate(vseries):
+        duplicates: Dict[str, List[Series]] = dict()
+        np.random.seed(self.settings.random_seed)
+
+        # define tiebreakers, which should have: name, value_func, pick_largest
```
Comment (snorthman) [inline src/picai_prep/dcm2mha.py:316]: This is an improvement, as is np.random.seed

File: src/picai_prep/dcm2mha.py [hunk 2]
```diff
@@ -277,44 +275,62 @@ def apply_mappings(self):
...
+        tiebreakers = [
+            ('slice count', lambda a: len(a.filenames), True),  # pick the series with the most slices
+            ('image resolution', lambda a: a.resolution, False),  # pick the series with the highest resolution
+        ]
```
Comment (snorthman) [inline src/picai_prep/dcm2mha.py:320]: I like listing the `tiebreakers`

File: src/picai_prep/dcm2mha.py [hunk 3]
```diff
@@ -213,13 +212,12 @@ def compile_log(self):
         log += ['\nSERIES', divider.replace('=', '-')]
         for i, serie in enumerate(self.series):
             log.append(f'({i}) {serie.path.as_posix()}')
-            log.extend([f'\t{l}' for l in serie._log])
+            log.extend([f'\t{item}' for item in serie._log])
             log.append(f'\tFATAL: {serie.error}\n' if not serie.is_valid else '')
         return '\n'.join(log)
 
-    def convert(self, *args):
+    def convert(self, output_dir):
```
[the remark this answers is not in the document]
Comment (joeranbosma) [reply src/picai_prep/dcm2mha.py:219]: Did not mean to upset you. In my mind the "Minor cleanup" part of the message covered this, after I accidentally bundled these changes with the refactor.

File: src/picai_prep/dcm2mha.py [hunk 4]
```diff
@@ -277,44 +275,62 @@ def apply_mappings(self):
 
         self.write_log(f'\t({plural(len(errors), "error")}{f" {errors}" if len(errors) > 0 else ""})')
 
+    def choose_best_series(self, all_series: List[Series], tiebreakers: List[Tuple[str, callable, bool]], mapping: str):
```
[the remark this answers is not in the document]
Comment (joeranbosma) [reply src/picai_prep/dcm2mha.py:278]: I see it the other way around. Breaking it into two function also breaks up the complexity in two: one part deals with the actual tie-breaks, the other part deals with logistics. 
`chosen_series` and `dropped_series` makes it easy to understand what happens on a line, without even requiring an understanding of the code. The `zip` functionality isn't a necessity, can be refactored.

**Comment by joeranbosma** [comment]
Overall length has indeed gone up, but I would argue that the time to comprehend this function has gone down.

**Pull request closed without merging**